core/src/lib.rs

```diff
@@ -596,18 +599,25 @@ pub fn forward_tokens_inner_internal<T: Into<TokenStream2>>(tokens: T) -> Result
     let parsed = parse2::<ForwardedTokens>(tokens.into())?;
     let target_path = parsed.target_path;
     let imported_tokens = parsed.item;
-    let combined_tokens = match parsed.extra {
-        Some(extra) => quote! {
-            #imported_tokens,
-            #extra
-        },
-        None => quote!(#imported_tokens),
-    };
-    Ok(quote! {
-        #target_path! {
-            #combined_tokens
-        }
-    })
+    let tokens_forwarded_keyword = keywords::__private_macro_magic_tokens_forwarded::default();
+    let pound = Punct::new('#', Spacing::Alone);
+    match parsed.extra {
+        // no extra, used by attr, so expand to attribute macro
```

Comment:
Yeah also if I remember correctly the main blocker to having a more general thing like a `tt` here is anything but an `ident` or a literal limits what types of position you can use forward_tokens stuff because of how decl macros work. This is the same reason I have to do that crazy manual interpolation of idents with :: instead of just a path in the export tokens macro. Extra as a string works in all positions.

Maybe a tuple like this would be better though:
```rust
("extra item 1", "extra item 2", "extra item 3", "extra item 4")
```

I don't think that would mess up the position capabilities at all, and at least then we wouldn't have to do string interpolation and escaping

Reply:
But yes I think it is fine to expand to a proc macro here